src/components/map-container.js

```diff
@@ -249,54 +276,64 @@ export default function MapContainerFactory(MapPopover, MapControl, Editor) {
         return null;
       }
       // if clicked something, ignore hover behavior
-      const objectInfo = clicked || hoverInfo;
       let layerHoverProp = null;
-      let position = {x: mousePosition[0], y: mousePosition[1]};
-
-      if (interactionConfig.tooltip.enabled && objectInfo && objectInfo.picked) {
-        // if anything hovered
-        const {object, layer: overlay} = objectInfo;
-
-        // deckgl layer to kepler-gl layer
-        const layer = layers[overlay.props.idx];
-
-        if (layer.getHoverData && layersToRender[layer.id]) {
-          // if layer is visible and have hovered data
-          const {
-            config: {dataId}
-          } = layer;
-          const {allData, fields} = datasets[dataId];
-          const data = layer.getHoverData(object, allData);
-          const fieldsToShow = interactionConfig.tooltip.config.fieldsToShow[dataId];
-
-          layerHoverProp = {
-            data,
-            fields,
-            fieldsToShow,
-            layer
-          };
-        }
-      }
+      let layerPinnedProp = null;
+      const position = {x: mousePosition[0], y: mousePosition[1]};
+      let pinnedPosition = {};
+
+      layerHoverProp = getLayerProp({
+        interactionConfig,
+        hoverInfo,
+        layers,
+        layersToRender,
+        datasets
+      });
+
+      const compareMode = interactionConfig.tooltip.config
+        ? interactionConfig.tooltip.config.compareMode
+        : false;
 
       if (pinned || clicked) {
         // project lnglat to screen so that tooltip follows the object on zoom
         const viewport = new WebMercatorViewport(mapState);
         const lngLat = clicked ? clicked.lngLat : pinned.coordinate;
-        position = this._getHoverXY(viewport, lngLat);
+        pinnedPosition = this._getHoverXY(viewport, lngLat);
+        layerPinnedProp = getLayerProp({
+          interactionConfig,
+          hoverInfo: clicked,
+          layers,
+          layersToRender,
+          datasets
+        });
+        if (layerHoverProp) {
+          layerHoverProp.primaryData = layerPinnedProp.data;
+          layerHoverProp.compareType = interactionConfig.tooltip.config.compareType;
+        }
       }
       return (
         <div>
-          <MapPopover
-            {...position}
-            layerHoverProp={layerHoverProp}
-            coordinate={
-              interactionConfig.coordinate.enabled && ((pinned || {}).coordinate || coordinate)
-            }
-            freezed={Boolean(clicked || pinned)}
-            onClose={this._onCloseMapPopover}
-            mapW={mapState.width}
-            mapH={mapState.height}
-          />
+          {(clicked || pinned) && (
```

Comment:
the logic here has become way too complicated clarify with varible name will be helpful
what is `clicked || pinned` and what is `(compareMode || (!clicked && !pinned))` can it be named to `hasTooltip` `hasComparisonTooltip`?

Reply:
Got it